Sources/GutenbergKit/Sources/EditorViewController.swift

```diff
@@ -125,15 +135,30 @@ public final class EditorViewController: UIViewController, GutenbergEditorContro
         }
     }
 
-    func injectEditorConfiguration() {
-        evaluate("""
+    private func getEditorConfiguration() -> WKUserScript {
+        let siteIDString = siteID?.stringValue ?? ""
+        let authType = siteID != nil ? "Bearer" : "Basic"
+        let jsCode = """
         window.GBKit = {
-            siteURL: '\(siteURL)',
-            authToken: '\(authToken)'
+            siteAPIURL: '\(siteAPIURL)',
+            authToken: '\(authType) \(authToken)',
+            apiPathModifier
         };
+        function apiPathModifier(path) {
+            const siteID = '\(siteIDString)';
+            if (path && siteID.length && !path.includes(`/sites/${siteID}/`)) {
+                return path
+                    .replace('/wp/v2/', `/wp/v2/sites/${siteID}/`)
+                    .replace('/wpcom/v2/', `/wpcom/v2/sites/${siteID}/`)
+                    .replace('/oembed/1.0/', `/oembed/1.0/sites/${siteID}/`);
+            }
+        }
```

Comment:
When I proposed passing a function, I didn't consider our current practice of running `JSON.stringify` on these values. Passing a function to that will nullify it. Not an urgent issue given local storage is only used for development purposes, but we might address it at a different time.

Reply:
Thanks for the suggestion!

I'm trying a different approach in https://github.com/wordpress-mobile/GutenbergKit/pull/15/commits/f268246891ae3a76be68c72a409e5f5ee64d40c2

Where a new param is added `siteApiNamespace` to be able to format the final URL in a more dynamic way. What do you think about that approach?